x/stakeibc/keeper/msg_server_register_host_zone.go

```diff
@@ -113,31 +113,31 @@ func (k msgServer) RegisterHostZone(goCtx context.Context, msg *types.MsgRegiste
 	// generate delegate account
 	// NOTE: in the future, if we implement proxy governance, we'll need many more delegate accounts
 	delegateAccount := types.FormatICAAccountOwner(chainId, types.ICAAccountType_DELEGATION)
-	if err := k.RegisterInterchainAccount(ctx, zone.ConnectionId, delegateAccount, appVersion); err != nil {
+	if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, zone.ConnectionId, delegateAccount, appVersion); err != nil {
```

Comment:
i think we shouldnot use ICAControllerKeeper RegisterInterchainAccount func because it will soon be deprecated. We have to use the ICAControllerKeeper msg_server to call that func. That why i have implement a new func that use the msg_server